doc/lsst.pipe.base/command-line-task-config-howto.rst

```diff
@@ -0,0 +1,141 @@
+
+.. _command-line-task-config-howto-show-subset:
+
+Viewing a subset of configurations
+----------------------------------
+
+To filter the :option:`--show config <--show>` output, include a search term with wildcard (``*``) characters.
+For example, this will show any configuration that includes the string ``calibration``:
+
+.. code-block:: bash
+
+   processCcd.py REPOPATH --output output --show config="*calibration*"
+
+.. _command-line-task-config-howto-config:
+
+How to set configurations with command-line arguments
+=====================================================
+
+Command-line tasks can be configured through a combination of two mechanisms: arguments on the command line (:option:`--config`) or through configuration files (:option:`--configfile`).
+In general, simple configurations can be made through the command line, while complex configurations and :ref:`subtask retargeting <command-line-task-retargeting-howto>` must done through configuration files (see :ref:`command-line-task-config-howto-configfile`).
+
+To change a configuration value on the command line, pass that configuration name and value to the :option:`--config` argument.
+For example, to set a configuration named ``skymap.projection`` to a value ``"TAN"``:
+
+.. code-block:: bash
+
+   task.py REPOPATH --output output --config skyMap.projection="TAN"
+
+You can provide multiple :option:`--config` arguments on the same command line or set multiple configurations with a single :option:`--config` argument:
+
+.. code-block:: bash
+
+   task.py REPOPATH --output output --config config1="value1" config2="value2"
+
+Only simple configuration values can be set through :option:`--config` arguments, such as:
+
+- **String values**. For example: ``--config configName="value"``.
+- **Scalar numbers**. For example: ``--config configName=2.5``.
+- **Lists of integers**. For example: ``--config intList=2,4,-87``.
+- **List of floating point numbers**. For example: ``--config floatList=3.14,-5.6e7``.
+- **Boolean values**. For example: ``--config configName=True configName2=False``.
+
+Specific types configurations you **cannot** perform with the :option:`--config` argument are:
+
+- You cannot :doc:`retarget a subtask <command-line-task-retargeting-howto>` specified by a `lsst.pex.config.ConfigurableField` (which is the most common case).
+- For items in registries, you can only specify values for the active (current) item.
+- You cannot specify values for lists of strings.
+- You cannot specify a subset of list.
+  You must specify all values at once.
+
+For these more complex configuration types you must use configuration files.
+
+.. _command-line-task-config-howto-configfile:
+
+How to use configuration files
+==============================
+
+You also provide configurations to a command-line task through a *configuration file*
+In fact, configuration files are Python modules; anything you can do in Python you can do in a configuration file.
+
```

Comment:
Missing a period for the first sentence.

This section would really benefit from an example configuration file text block, even if it's just a few lines long.